src/main/java/core/controller/AuthenticationController.java

```diff
@@ -0,0 +1,27 @@
+package core.controller;
+
+import core.model.dto.UserRequestDto;
+import core.security.AuthenticationService;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class AuthenticationController {
+    private final AuthenticationService authenticationService;
+    
+    public AuthenticationController(AuthenticationService authenticationService) {
+        this.authenticationService = authenticationService;
+    }
+    
+    @PostMapping("/register")
+    public String registerUser(@RequestBody UserRequestDto requestDto) {
+        try {
+            authenticationService.register(requestDto.getEmail(), requestDto.getPassword());
+        } catch (Exception e) {
+            return "The user with specified email already exists. "
```

Comment:
Is it a good practice to return a String message in case of error? I don't think so.
One more - you should not catch exceptions in controllers. Rather handle them on the service layer or add to method signature in controllers.

Reply:
Sure, I know it isn't a proper way to do this. Just a simple test for me.